tests/xoto3/dynamodb/write_versioned/transact/modify_test.py

```diff
@@ -12,3 +19,37 @@ def test_cant_delete_non_prefetched_item_without_specifying_key():
 
     with pytest.raises(TableSchemaUnknownError):
         delete(tx, "table2", dict(id=4, value=7, other_value=9))
+
+
+def test_presume():
+    tx = VersionedTransaction(dict())
+    tx = presume(tx, "steve", dict(group="g", id="123"), None)
+    assert None is get(tx, "steve", dict(group="g", id="123"))
+
+
+def test_presume_already_exists():
+    key = dict(id=123)
+    tx = VersionedTransaction(
+        dict(
+            steve=_TableData(
+                items={hashable_key(key): dict(key, foo=2)}, effects=dict(), key_attributes=("id",)
+            )
+        )
+    )
+    tx = presume(tx, "steve", key, None)
+    assert require(tx, "steve", dict(id=123))["foo"] == 2
+
+
+def test_define_table():
+    tx = VersionedTransaction(dict())
+    tx = define_table(tx, "BobTable", "group", "id")
```

Comment:
Did you consider this?
```suggestion
    tx = define_table("BobTable", "group", "id")(tx)
```

Reply:
I suppose there is less reason someone would need to `flow` a table definition.

Reply:
define_table is the low-level API - it's just following convention by taking the same shape as everything down there.

you can do the flow stuff with the TypedTable.define approach.